apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/TransactionPool.cs

```diff
@@ -126,6 +115,18 @@ public static TransactionOptions GetLastUsedTransactionOptions(this Session sess
             return null;
         }
 
+        public static Transaction GetCachedTransaction(this Session session,
+            TransactionOptions options)
+        {
+            SessionInfo info;
+            if (s_sessionInfoTable.TryGetValue(session, out info)
+                && info.ActiveTransactionOptions != null
+                && Equals(info.ActiveTransactionOptions, options))
```

Comment:
Since you know `info.ActiveTransactionOptions` is non-null, this could alternatively be `info.ActiveTransactionOptions.Equals(options)`.